transaction-service/src/main/java/pl/adamboguszewski/transaction/service/application/Transaction.java

```diff
@@ -20,40 +20,55 @@
     UUID transactionId;
     @Column
     Long totalPrice;
-
     @Column
-    @OneToOne(mappedBy = "transaction")
+    LocalDateTime transactionDateTime;
+
+//    @Column
+    @OneToOne(
+        mappedBy = "transaction",
+        orphanRemoval = true,
+        cascade = CascadeType.ALL)
     TransactionInformation transactionInformation;
 
-    @OneToMany(mappedBy = "transaction", cascade = CascadeType.ALL)
-    List<TransactionProduct> products;
+    @OneToMany(
+            mappedBy = "transaction",
+            orphanRemoval = true,
+            cascade = CascadeType.ALL,
+            fetch = FetchType.EAGER)
+    Set<TransactionProduct> products;
```

Comment:
Databases works better with sets instead of lists, please read about HashCode, Equals, Set, HashSet, Map, HashMap